chain-impl-mockchain/src/testing/e2e/fees.rs

```diff
@@ -239,3 +244,93 @@ fn verify_total_funds_after_transaction_with_fee(fee: u64) {
         .pots()
         .has_fee_equals_to(&Value(fee));
 }
+
+#[quickcheck]
+pub fn vote_cast_fees(constant_fee: u64, coefficient: u64, certificate: u64) {
+    let expected_fees = constant_fee + coefficient + certificate;
+    let favorable = Choice::new(1);
+    let voting_token = TokenName::try_from(vec![0u8; TOKEN_NAME_MAX_SIZE]).unwrap();
+
+    let (mut ledger, controller) = prepare_scenario()
+        .with_config(
+            ConfigBuilder::new()
+                .with_fee(LinearFee::new(constant_fee, coefficient, certificate))
+                .with_rewards(Value(1000)),
+        )
+        .with_initials(vec![wallet(ALICE)
+            .with(1_000)
+            .with_token(voting_token, 1_000)
+            .owns(STAKE_POOL)
+            .committee_member()])
+        .with_vote_plans(vec![vote_plan(VOTE_PLAN)
+            .owner(ALICE)
+            .consecutive_epoch_dates()
+            .with_proposal(
+                proposal(VoteTestGen::external_proposal_id())
+                    .options(3)
+                    .action_off_chain(),
+            )])
+        .build()
+        .unwrap();
+
+    let mut alice = controller.wallet(ALICE).unwrap();
+    let vote_plan = controller.vote_plan(VOTE_PLAN).unwrap();
+    let proposal = vote_plan.proposal(0);
+
+    controller
+        .cast_vote_public(&alice, &vote_plan, &proposal.id(), favorable, &mut ledger)
+        .unwrap();
+    alice.confirm_transaction();
+
+    LedgerStateVerifier::new(ledger.clone().into())
+        .info("fee pot is filled with expected fees amount")
+        .pots()
+        .has_fee_equals_to(&Value(expected_fees));
+
+    LedgerStateVerifier::new(ledger.into())
+        .info("account balance is correct")
+        .address_has_expected_balance(alice.as_account_data(), Value(1000 - expected_fees));
```

Comment:
It would be better to assign some variable to express 1000. Because when reading test it's not 100% clear that this 1000 is initial account balance or initial rewards

Reply:
And it would also be better to have different values for :

- rewards
- voting tokens
- balance

to avoid any confusion